system/security/shared/caching.cpp

```diff
@@ -714,47 +835,64 @@ SecAccessFlags CPermissionsCache::queryDefaultPermission(ISecUser& user)
     }
 
     if (addedToCache)
-    {
         DBGLOG("Added user '%s' to default file scope permissions with access %s(%d)", username.c_str(), getSecAccessFlagName(defaultPermission),
-               defaultPermission);
-    }
+                    defaultPermission);
 
     return defaultPermission;
 }
 
 void CPermissionsCache::flush()
 {
-    // MORE - is this safe? m_defaultPermossion and m_lastManagedFileScopesRefresh are unprotected,
-    // and entries could be added to the first cache while the second is being cleared - does that matter?
-    {
-        WriteLockBlock writeLock(m_resPermCacheRWLock);
-        MapResPermissionsCache::const_iterator i;
-        MapResPermissionsCache::const_iterator iEnd = m_resPermissionsMap.end();
-        for (i = m_resPermissionsMap.begin(); i != iEnd; i++)
-            delete (*i).second;
-        m_resPermissionsMap.clear();
-    }
-    {
-        WriteLockBlock writeLock(m_userCacheRWLock );
-        MapUserCache::const_iterator ui;
-        MapUserCache::const_iterator uiEnd = m_userCache.end();
-        for (ui = m_userCache.begin(); ui != uiEnd; ui++)
-            delete (*ui).second;
-        m_userCache.clear();
-    }
+    clearPermissionsCache();
```

Comment:
What happens if this function is called at the same time as permissions are being resolved?  You could get inconsistent values.